services/director-v2/src/simcore_service_director_v2/core/dynamic_services_settings/sidecar.py

```diff
@@ -50,6 +48,55 @@ def enforce_r_clone_requirement(cls, v: int, values) -> PositiveInt:
         return v
 
 
+class PlacementConstraintStr(ConstrainedStr):
+    strip_whitespace = True
+    regex = re.compile(
+        r"^(?!-)(?![.])(?!.*--)(?!.*[.][.])[a-zA-Z0-9.-]*(?<!-)(?<![.])(!=|==)[a-zA-Z0-9_. -]*$"
+    )
+
+
+class PlacementSettings(BaseCustomSettings):
+    # This is just a service placement constraint, see
+    # https://docs.docker.com/engine/swarm/services/#control-service-placement.
+    DIRECTOR_V2_SERVICES_CUSTOM_CONSTRAINTS: list[PlacementConstraintStr] = Field(
+        default_factory=list,
+        example='["node.labels.region==east", "one!=yes"]',
+    )
+
+    DIRECTOR_V2_PLACEMENT_CONSTRAINTS_REPLACEMENTS_FOR_GENERIC_RESOURCES: dict[
+        str,
+        PlacementConstraintStr,
+    ] | None = Field(
```

Comment:
I think you would probably spare some headaches if instead of allowing None, you would go for just the dictionary.
then you just replace whatever the keys of the dictionary are. Also in case we start re-using them this will not conflict with this code.

so for example:
- {} would replace nothing
- {"AIRAM": "node.labels.airam==true"} would only replace AIRAM but NOT VRAM
- {"AIRAM": "node.labels.airam==true", "VRAM": "node.labels.vram==true"} would replace both AIRAM and VRAM